File: day2/day2.py
from advent import Session

aoc = Session(2021,2)
silver = None
gold = None

def parse(line):
    direction,dist = line.strip().split()
    return (direction,int(dist))

with aoc.fp() as f:
    lines = [parse(line) for line in f.readlines()]

horz,depth,aim,depth2 = 0,0,0,0
for direction,dist in lines:
    if direction == "forward":
        horz += dist
        depth2 += aim * dist
    elif direction == "down":
        depth += dist
        aim += dist
    elif direction == "up":
        depth -= dist
        aim -= dist

silver = horz*depth
gold = horz*depth2
print(silver,gold)

aoc.solution(1,silver)
aoc.solution(2,gold)

File: day2/Day2.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Program {
    public static int Main(string[] args) {
        List<string> lines = System.IO.File.ReadLines("day2.input").ToList();
        List<(string,int)> commands = (from string line in lines
            select (line.Split(" ")[0],int.Parse(line.Split(" ")[1]))).ToList();
    
        int depth = 0;
        int horizontal = 0;
        int aim = 0;
        int depth_part_2 = 0;

        foreach((string direction, int dist) cmd in commands) {
            switch(cmd.direction) {
                case "forward":
                        horizontal += cmd.dist;
                        depth_part_2 += aim*cmd.dist;
                        break;
                case "up":
                        depth -= cmd.dist;
                        aim -= cmd.dist;
                        break;
                case "down":
                        depth += cmd.dist;
                        aim += cmd.dist;
                        break;
                default:
                        Console.WriteLine("Unknown direction '{0}' in command",cmd.direction);
                        break;
            };
        }

        Console.WriteLine("Silver: {0}",horizontal*depth);
        Console.WriteLine("Gold: {0}",horizontal*depth_part_2);

        return 0;
    }
}
